torchci/components/JobAnnotationToggle.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { JobData } from "../lib/types";
+import { ToggleButtonGroup, ToggleButton } from "@mui/material";
+
+type JobAnnotation =
+  | "infra_flake"
+  | "time_out"
+  | "SEV"
+  | "broken_trunk"
+  | "test_flake"
+  | "null";
+
+const testFailureRe = /^(?:FAIL|ERROR) \[.*\]: (test_.* \(.*Test.*\))/;
+
+export default function JobAnnotationToggle({ job }: { job: JobData }) {
+  const [state, setState] = React.useState<JobAnnotation>(
+    (job.failureAnnotation ?? "null") as JobAnnotation
+  );
+
+  async function handleChange(
+    _: React.MouseEvent<HTMLElement>,
+    newState: JobAnnotation
+  ) {
+    setState(newState);
+    await fetch(`/api/job_annotation/${job.repo}/${job.id}/${newState}`, {
+      method: "POST",
+    });
+  }
+
+  // Don't show this button if we know this is a test failure, since we do not
+  // consider test failures to be infra failures.
+  const match = job.failureLine != null && job.failureLine.match(testFailureRe);
+  if (match !== null) {
```

Comment:
Let's remove this bit. Reasons:

1. This results in failures getting skipped even when an infra issue causes the logs to not identify a failure line
2. Better to explicitly call out a test that fails consistently in trunk than to not categorize the failure at all